distributed/worker.py

```diff
@@ -1475,17 +1476,13 @@ def add_task(
                     # This is a scheduler re-assignment
                     # Either `fetch` -> `waiting` or `flight` -> `waiting`
                     self.log.append((ts.key, "re-adding key, new TaskState"))
-                    self.transition(ts, "waiting", runspec=runspec)
+                    recommendations.update({ts: ("waiting", {"runspec": runspec})})
```

Comment:
This pattern is in a lot of places -- where previously we were passing through `kwargs` via `transition`, now they're bundled as a separate dictionary that gets passed in to `self.transition` by `self.transitions` while iterating over the recommended transitions.